### PR TITLE
Add support for UID MOVE (RFC 6851)

### LINKED ISSUE #142: MOVE support
Originally reported by: **Menno Smits (Bitbucket: [mjs0](https://bitbucket.org/mjs0))**

---

And fallback when the server doesn't have it as per:
http://stackoverflow.com/a/15816045/195383

---
- Bitbucket: https://bitbucket.org/mjs0/imapclient/issue/145

### PR DESCRIPTION
**Pull request by NicolasLM**
Allows to move messages in a single call without going through
copy + flag deleted + expunge.

Fixes #142

### CODE REVIEW

**Comment by NicolasLM** [comment]
I couldn't get the livetests to run on the account I've found having the `MOVE` capability.

**Comment by mlorant** [comment]
I can't find any provider with the MOVE capability with my personal accounts... (Does Gmail supports it? Have not checked using the XOAUTH2 mecanism)

The code seems pretty straight-forward, but i can't test it :(

File: doc/src/releases.rst [hunk 1]
```diff
@@ -14,6 +14,8 @@ Changed
   servers.
 - More precise exceptions available in `imapclient.exceptions` are raised when
   an error happens
+- Atomically move messages to another folder using the MOVE extension
+  (RFC 6851)
```
Comment (mlorant) [inline doc/src/releases.rst]: ```(:rfc:`6851`)``` for pretty automatic link.

Comment (NicolasLM) [reply doc/src/releases.rst]: 👍 done

File: imapclient/imapclient.py [hunk 2]
```diff
@@ -1166,6 +1169,16 @@ def copy(self, messages, folder):
                                        self._normalise_folder(folder),
                                        uid=True, unpack=True)
 
+    def move(self, messages, folder):
+        """Atomically move messages to another folder.
+
+        Requires the MOVE capability, see :rfc:`6851`.
+        """
```
Comment (mjs) [inline imapclient/imapclient.py]: It might be worth mentioning that `messages` must contain UIDs, regardless of what `use_uid` is set to.

**Comment by mjs** [comment]
I'm pretty sure Dovecot supports MOVE and maybe Fastmail too. I can try and test this.

File: imapclient/imapclient.py (hunk 2 above)
Comment (NicolasLM) [reply imapclient/imapclient.py]: Done

**Comment by NicolasLM** [comment]
@mjs Did you manage to try this code?

**Pull request merged**